File: day-1-2.rb
file = File.open("./inputs/day-1.txt")
file_data = file.readlines.map(&:chomp)
file.close()

increases = 0
lastValue = 0
file_data.each_with_index do |number, index|
    sum = number.to_i +  file_data[index+1].to_i +  file_data[index +2].to_i
    if sum > lastValue and lastValue > 0
        increases += 1 
    end 

    lastValue = sum;
end

print increases

File: day-1-2.mjs
import { promises as fs } from "fs";

async function loadData() {
  return (await fs.readFile("inputs/day-1.txt")).toString();
}

const data = (await loadData()).split("\n");

const result = data.reduce(
  (payload, next, index) => {
    const sum = +next + +data[index + 1] + +data[index + 2];
    sum > payload.lastValue && payload.increases++;
    payload.lastValue = sum;
    return payload;
  },
  { lastValue: 0, increases: -1 }
);

console.log(result.increases);
